### DISCUSSION TITLE
Best practice App Bindings in combination with Input Widget

### DISCUSSION THREAD

**Discussion by larsagny** [Textualize/textual#3480] [Q&A] [answered]
I'm wondering how to let a footer menu play nicely together with a input widget. My main app screen has a input widget in the center of the screen as a main entry point (like search bar). In the footer are a couple of bindings, like the `V=version` and `Q=Quit`.  I can make sure the binding is executed by `Binding(key="v", action="app.push_screen('version')", description="Version", priority=True)`. This make the input unusable as a user never will be able to type something that contains a `v`. So before executing the binding the input widget has to loose focus but I'm wondering what would be a good strategy here? It's not very intuitive to press TAB before hitting the `v` key.

**Comment by davep** [comment, marked as answer]
I'm a little confused by your question; you seem to be saying that you want the user to be able to press <kbd>v</kbd> and have it run the action that pops up the version screen, but you *also* want them to be able to type a "v" in the `Input`, but you also don't want them to have to move focus out of the `Input` to do this.

These things seem mutually exclusive.

Could you clarify that?

More generally though: this is a reason why it would generally be a bad idea to bind a letter or number key, app-wide, to some action. It would normally make more sense to bind a key with a modifier; <kbd>Ctrl</kbd>+<kbd>v</kbd> for the version screen, perhaps, or even simply a function key?

**Reply by larsagny** [reply]
Thanks for the quick reply. I was a little confused when I turned here to ask the question. I designed my footer menu based on the [screen docs](https://textual.textualize.io/guide/screens/). As all screen bindings there use single letters, I took that over. My main screen doesn't have an input field, just the landing screen which then wouldn't work anymore. Later I figured as well, that I should stick to something like `Ctrl+v` and rather refactor the footer bindings than the other way around.
